addons/hr_expense/models/hr_expense.py

```diff
@@ -217,22 +249,35 @@ def _compute_is_ref_editable(self):
             else:
                 expense.is_ref_editable = is_account_manager
 
+    @api.depends('product_id')
+    def _compute_product_description(self):
+        self.product_description = ""
+        for expense in self.filtered('product_id'):
+            expense.product_description = expense.product_id.description if not is_html_empty(html_sanitize(expense.product_id.description, strip_style=True, strip_classes=True)) else False
+
     @api.depends('product_id', 'company_id')
     def _compute_from_product_id_company_id(self):
         for expense in self.filtered('product_id'):
             expense = expense.with_company(expense.company_id)
             expense.name = expense.name or expense.product_id.display_name
             expense.product_uom_id = expense.product_id.uom_id
-            if not expense.attachment_number or (expense.attachment_number and not expense.unit_amount):
-                expense.unit_amount = expense.product_id.price_compute(
-                    'standard_price',
-                    uom=expense.product_uom_id,
-                    currency=expense.currency_id)[expense.product_id.id]
             expense.tax_ids = expense.product_id.supplier_taxes_id.filtered(lambda tax: tax.company_id == expense.company_id)  # taxes only from the same company
             account = expense.product_id.product_tmpl_id._get_product_accounts()['expense']
             if account:
                 expense.account_id = account
 
+    @api.depends('product_id', 'company_id')
+    def _compute_unit_amount(self):
+        for expense in self.filtered('product_id'):
+            if not expense.attachment_number or (expense.attachment_number and not expense.unit_amount):
+                if expense.product_has_cost:
+                    expense.unit_amount = expense.product_id.price_compute(
+                        'standard_price',
+                        uom=expense.product_uom_id,
+                        currency=expense.currency_id)[expense.product_id.id]
+                if not expense.product_has_cost: # and not expense.unit_amount:
+                    expense.unit_amount = expense.total_amount_company
```

Comment:
To avoid looping 1 time over all the records before doing anything, I suggest:
```suggestion
    @api.depends('product_id', 'company_id')
    def _compute_unit_amount(self):
        for expense in self:
            if not expense.product_id or (expense.attachment_number and expense.unit_amount):
                continue
            if expense.product_has_cost:
                expense.unit_amount = expense.product_id.price_compute(
                    'standard_price',
                    uom=expense.product_uom_id,
                    currency=expense.currency_id)[expense.product_id.id]
            else:
                expense.unit_amount = expense.total_amount_company
```
This could be done on ``_compute_from_product_id_company_id`` too